maintenance/log/sink.go

```diff
@@ -45,13 +47,27 @@ func SinkContextTransfer(sourceCtx, targetCtx context.Context) context.Context {
 // logs, created with log.Ctx(ctx), inside the context
 // and use them at a later point in time
 type Sink struct {
-	Silent       bool
-	jsonLogLines []string
+	Silent     bool
+	CustomSize int
+	ring       *stringRing
```

Comment:
I think we should allocate the ring on the stack and avoid the pointer, that should then also prevent the init

Reply:
I can remove the pointer, although that may not necessarily allocate it from the stack. The init on the other hand is for ensuring thread safety, and for enabling us to improve on the functionality in the future without having to work around it, like we do now.

A sink should be initialized with a certain size which is not allowed to change after creation.